cpp/src/io/parquet/decode_preprocess.cu

```diff
@@ -350,7 +461,15 @@ CUDF_KERNEL void __launch_bounds__(preprocess_block_size)
   // retrieve total string size.
   // TODO: make this block-based instead of just 1 warp
   if (compute_string_sizes) {
-    if (t < 32) { s->page.str_bytes = gpuDecodeTotalPageStringSize(s, t); }
+    size_type str_bytes = 0;
+    if (s->page.encoding == Encoding::DELTA_BYTE_ARRAY) {
+      // this must be called by all threads
+      str_bytes = gpuDeltaPageStringSize(s, t);
+    } else if (t < warp_size) {
+      // single warp
+      str_bytes = gpuDecodeTotalPageStringSize(s, t);
+    }
+    if (t == 0) { s->page.str_bytes = str_bytes; }
```

Comment:
Suggest refactoring gpuDecodeTotalPageStringSize as the single entry point and doing this check inside of it.